rtc/hal_rtc_nrf5x.c

```diff
@@ -140,150 +128,150 @@ void hal_rtc_init(void)
-{
-	uint32_t ctx;
-	struct hal_rtc_timer *entry;
-    
-	if ((timer == NULL) || (timer->link.tqe_prev != NULL) ||
-			(timer->cb_fun == NULL)) {
-		return false;
-	}
-    
-	timer->expiry = tick;
-
-	__HAL_DISABLE_INTERRUPTS(ctx);
-
-	if (TAILQ_EMPTY(&hal_timer_q)) {
-		TAILQ_INSERT_HEAD(&hal_timer_q, timer, link);
-	} else {
-		TAILQ_FOREACH(entry, &hal_timer_q, link) {
-			if ((int32_t)(timer->expiry - entry->expiry) < 0) {
-				TAILQ_INSERT_BEFORE(entry, timer, link);
-				break;
-			}
-		}
-		
-		if (!entry) {
-			TAILQ_INSERT_TAIL(&hal_timer_q, timer, link);
-		}
-	}
-
-	/* If this is the head, we need to set new OCMP */
-	if (timer == TAILQ_FIRST(&hal_timer_q)) {
-		set_ocmp(timer->expiry);
-	}
-
-	__HAL_ENABLE_INTERRUPTS(ctx);
-}
-
-
-void hal_rtc_stop(struct hal_rtc_timer *timer)
-{
-	uint32_t ctx;
-	int reset_ocmp;
-	struct hal_rtc_timer *entry;
-
-	if (timer == NULL) {
-		return;
-	}
-
-	__HAL_DISABLE_INTERRUPTS(ctx);
-
-	if (timer->link.tqe_prev != NULL) {
-		reset_ocmp = 0;
-		if (timer == TAILQ_FIRST(&hal_timer_q)) {
-			/* If first on queue, we will need to reset OCMP */
-			entry = TAILQ_NEXT(timer, link);
-			reset_ocmp = 1;
-		}
-
-		TAILQ_REMOVE(&hal_timer_q, timer, link);
-		timer->link.tqe_prev = NULL;
-		if (reset_ocmp) {
-			if (entry) {
-				set_ocmp(entry->expiry);
-			} else {
-				disable_ocmp();
-			}
-		}
-	}
-
-	__HAL_ENABLE_INTERRUPTS(ctx);
-}
-
-
-void rtc0_handler(void)
-{
-	uint32_t compare, overflow;
-	compare = NRF_RTC0->EVENTS_COMPARE[0];
-	
-	if (compare) {
-		NRF_RTC0->EVENTS_COMPARE[0] = 0;
-	}
-
-	overflow = NRF_RTC0->EVENTS_OVRFLW;
-	if (overflow) {
-		NRF_RTC0->EVENTS_OVRFLW = 0;
-		counter_high += (1UL << 24);
-	}
-
-	chk_queue();
-
-	/* Recommended by nordic to make sure interrupts are cleared */
-	compare = NRF_RTC0->EVENTS_COMPARE[0];
-}
+//
+//void hal_rtc_deinit(void)
+//{	
+//	NRF_RTC0->TASKS_STOP = 1;
+//	NVIC_DisableIRQ(RTC0_IRQn);
+//	NVIC_ClearPendingIRQ(RTC0_IRQn);
+//
+//        NRF_CLOCK->TASKS_LFCLKSTOP = 1;
+//}
+//
+//
+//uint32_t hal_rtc_time(void)
+//{
+//	uint32_t low32;
+//	uint32_t ctx;
+//	uint32_t tcntr;
+//	__HAL_DISABLE_INTERRUPTS(ctx);
+//	tcntr = counter_high;
+//	low32 = NRF_RTC0->COUNTER;
+//
+//	if (NRF_RTC0->EVENTS_OVRFLW) {
+//		tcntr += (1UL << 24);
+//		counter_high = tcntr;
+//		low32 = NRF_RTC0->COUNTER;
+//		NRF_RTC0->EVENTS_OVRFLW = 0;
+//		NVIC_SetPendingIRQ(RTC0_IRQn);
+//	}
+//	
+//	tcntr |= low32;
+//	__HAL_ENABLE_INTERRUPTS(ctx);
+//
+//	return tcntr;
+//}
+//
+//
+//bool hal_rtc_start(struct hal_rtc_timer *timer, uint32_t ticks)
+//{
+//	uint32_t tick;
+//        tick = hal_rtc_time() + ticks;
+//	return hal_rtc_start_at(timer, tick);
+//}
+//
+//
+//bool hal_rtc_start_at(struct hal_rtc_timer *timer, uint32_t tick)
+//{
+//	uint32_t ctx;
+//	struct hal_rtc_timer *entry;
+//    
+//	if ((timer == NULL) || (timer->link.tqe_prev != NULL) ||
+//			(timer->cb_fun == NULL)) {
+//		return false;
+//	}
+//    
```

Comment:
Is all this code supposed to be commented away?

Reply:
No, It does not. I commented them away just to test if the device's firmware use it, and to show you that we are currently not using them. So, Do you want to restore those code ?